frontend/src/screens/wallet/send/Onchain.tsx

```diff
@@ -0,0 +1,234 @@
+        title: "Failed to send payment",
+        description: "" + e,
+      });
+      console.error(e);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  React.useEffect(() => {
+    if (!address) {
+      navigate("/wallet/send");
+    }
+  }, [navigate, address]);
+
+  if (!balances || !address || (!recommendedFees && !mempoolError)) {
+    return <Loading />;
+  }
+
+  return (
+    <form onSubmit={onSubmit} className="grid gap-6 md:max-w-lg">
+      <div className="grid gap-2">
+        <Label>Recipient</Label>
+        <div className="flex flex-wrap gap-2 items-center">
+          {address.match(/.{1,4}/g)?.map((word, index) => {
+            if (index % 2 === 0) {
+              return (
+                <span key={index} className="text-foreground">
+                  {word}
+                </span>
+              );
+            } else {
+              return (
+                <span key={index} className="text-muted-foreground">
+                  {word}
+                </span>
+              );
+            }
+          })}
+        </div>
+      </div>
+      <div className="grid gap-2">
+        <Label htmlFor="amount">Amount</Label>
+        <Input
+          id="amount"
+          type="number"
+          value={amount}
+          placeholder="Amount in Satoshi..."
+          onChange={(e) => {
+            setAmount(e.target.value.trim());
+          }}
+          min={isSwap ? 50000 : 1}
+          required
+          autoFocus
+          className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none"
+          endAdornment={
+            <FormattedFiatAmount amount={Number(amount)} className="mr-2" />
+          }
+        />
+        <div className="grid">
+          <div className="flex justify-between text-muted-foreground text-xs sensitive slashed-zero">
+            <div>
+              {isSwap ? "Spending Balance: " : "On-chain Balance: "}
+              {new Intl.NumberFormat().format(
+                isSwap
+                  ? balances.lightning.totalSpendable / 1000
+                  : balances.onchain.spendable
+              )}{" "}
+              sats
+            </div>
+            <FormattedFiatAmount
+              className="text-xs"
+              amount={
+                isSwap
+                  ? balances.lightning.totalSpendable / 1000
+                  : balances.onchain.spendable
+              }
+            />
+          </div>
+          {isSwap && (
+            <div className="flex justify-between text-muted-foreground text-xs sensitive slashed-zero">
+              <div>Minimum: 50000 sats</div>
+              <FormattedFiatAmount className="text-xs" amount={50000} />
+            </div>
+          )}
+        </div>
+      </div>
+      <div className="flex items-center justify-between">
+        <Label htmlFor="swap" className="font-medium text-sm cursor-pointer">
+          Swap from Spending Balance
```

Comment:
when this switch is off, I think it would be cool to show the spending balance below the switch

Reply:
hmm. The problem I see right now we can't see the balance until we toggle the switch.

Or could we add it to "Swap from Spending Balance" e.g. "Swap from Spending Balance (300k sats)"

Reply:
I did that, but I still feel what we have already is better:

https://github.com/user-attachments/assets/e3d18176-85cc-4602-b81f-188b9cd534f4

Reply:
hmm. Yeah it is kind of ugly like this. @stackingsaunter @reneaaron any thoughts?